spring-integration-ftp/src/main/java/org/springframework/integration/ftp/session/FtpSession.java

```diff
@@ -71,7 +74,17 @@ public boolean remove(String path) throws IOException {
 
 	@Override
 	public FTPFile[] list(@Nullable String path) throws IOException {
-		return this.client.listFiles(path);
+		FTPFile[] ftpFiles = this.client.listFiles(path);
+		if (StringUtils.hasText(path)) {
+			List<EnhancedFTPFile> enhancedFtpFiles = new LinkedList<>();
```

Comment:
Optimization: Since we know the length of the array from line 77 couldn't we allocate an array here and remove the use of the linked list.   Nitpick.